coherent_rates/isf.py

```diff
@@ -185,11 +189,110 @@ def get_boltzmann_isf(
     for _i in range(n_repeats):
         state = _get_random_boltzmann_state_from_hamiltonian(
             hamiltonian,
-            config.temperature,
+            temperature,
         )
         data = _get_isf_from_hamiltonian(hamiltonian, operator, state, times)
         isf_data += data["data"]
     return {
         "data": isf_data / n_repeats,
         "basis": times,
     }
+
+
+def get_boltzmann_isf(
+    system: PeriodicSystem,
+    config: PeriodicSystemConfig,
+    times: _AX0Inv,
+    direction: tuple[int] = (1,),
+    *,
+    n_repeats: int = 10,
+) -> ValueList[_AX0Inv]:
+    hamiltonian = get_hamiltonian(system, config)
+    return get_boltzmann_isf_from_hamiltonian(
+        hamiltonian,
+        config.temperature,
+        times,
+        direction,
+        n_repeats=n_repeats,
+    )
+
+
+def get_boltzmann_isf_stats(
+    system: PeriodicSystem,
+    config: PeriodicSystemConfig,
+    times: _AX0Inv,
+    direction: tuple[int] = (1,),
+    *,
+    average_over: int = 10,
+) -> StatisticalValueList[_AX0Inv]:
+    isf_data = np.zeros((average_over, times.n), dtype=np.complex128)
+    hamiltonian = get_hamiltonian(system, config)
+    operator = get_periodic_x_operator(hamiltonian["basis"][0], direction)
+
+    for _i in range(average_over):
+        state = _get_random_boltzmann_state_from_hamiltonian(
+            hamiltonian,
+            config.temperature,
+        )
+        data = _get_isf_from_hamiltonian(hamiltonian, operator, state, times)
+        isf_data[_i, :] = data["data"]
+    mean = np.mean(isf_data, axis=0, dtype=np.complex128)
+    sd = np.std(isf_data, axis=0, dtype=np.complex128)
+    return {
+        "data": mean,
+        "basis": times,
+        "standard_deviation": sd,
+    }
+
+
+def truncate_value_list(
+    values: ValueList[EvenlySpacedTimeBasis[int, int, int]],
+    index: int,
+) -> ValueList[EvenlySpacedTimeBasis[int, int, int]]:
+    data = values["data"][0 : index + 1]
+    new_times = EvenlySpacedTimeBasis(index + 1, 1, 0, values["basis"].times[index])
+    return {
+        "basis": new_times,
+        "data": data,
+    }
+
+
+def get_ak_data_1d(
+    system: PeriodicSystem,
+    config: PeriodicSystemConfig,
+    times: _AX0Inv,
+    *,
+    n_points: int = 8,
+) -> tuple[
```

Comment:
Should return a ValueList[DeltaKBasis] where DeltaKBasis is just a FundamentalBasis with the delta_k array stored inside it.